app/models/user.rb

```diff
@@ -376,19 +379,23 @@ def get_user_otp_at time
   def user_passwd_response
     user_hash = {}
     user_hash[:pw_name] = user_login_id
-    user_hash[:pw_passwd]  = "x"
+    user_hash[:pw_passwd] = 'x'
     user_hash[:pw_uid] = uid.to_i
 
     # If gid is supplied (avoid N+1)
-    if self.respond_to?(:gid) && gid
+    if respond_to?(:gid) && gid
       user_hash[:pw_gid] = gid.to_i
-    else
-      user_hash[:pw_gid] = groups.where(name: user_login_id).first.gid if groups.where(name: user_login_id).count > 0
+    elsif respond_to?(:gid_count)
+      if gid_count.positive?
+        user_hash[:pw_gid] = groups.where(name: user_login_id).first.gid
+      end
+    elsif groups.where(name: user_login_id).count.positive?
```

Comment:
this part, isn't this also do multiple of the same query?

Reply:
This part only happen when:
`gid is not supplied or nil` and
`gid_count is not supplied`

This happen when this method is called by method other than `get_sysadmins`